### PR TITLE
chore: update Avatar to use makeResetStyles

### PR DESCRIPTION
**Pull request by spmonahan**
## Previous Behavior

Avatar had many CSS classes on its slots.
## New Behavior

Number of CSS classes is reduced.

New class counts. In cases where there are multiple examples for a story I always picked the first example.

The two outliers here are the badge and active stories. In the case of badge a `PresenceBadge` is ultimately rendered in the slot and that control is not using `makeResetStyles`. Updating `PresenceBadge` will bring the badge numbers down.

The active styles use the `::before` pseudo element and styles for this element are conditionally applied for these styles. The overall number of class names could be brought down significantly by rolling these styles into the base reset styles however we'd then be rendering `::before` in all cases which will increase the DOM and style weight of the Avatar in all cases (in particular in the default case). Since only a small number of Avatars should ever be active at once the relatively large number of class names in this instance should not be an issue.

| Story | Old | New |
| --- | --- | --- |
| default | 54 | 14 |
| name | 49 | 8 |
| image | 62 | 10 |
| icon | 54 | 14 |
| badge | 76 | 32 |
| badgeIcon | 76 | 32 |
| square | 54 | 15 |
| colorBrand | 49 | 8 |
| colorColorful | 49 | 8 |
| colorPalette | 49 | 8 |
| active | 81 | 40 |
| activeAppearance | 81 | 40 |
| initials | 49 | 8 |
| size | 49 | 12 |

## Related Issues

- #26814
- #26817

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Avatar-mount_base.html">1558</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Avatar-mount.html">1648</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Button-mount_base.html">1138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Button-mount.html">1138</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Field-mount_base.html">2527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Field-mount.html">2464</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProvider-mount_base.html">1982</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProvider-mount.html">1933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-mount_base.html">739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-mount.html">759</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">709</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">700</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">777</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">753</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/InfoButton-mount_base.html">671</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/InfoButton-mount.html">674</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/MakeStyles-mount_base.html">2230</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/MakeStyles-mount.html">2255</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Persona-mount_base.html">3834</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/Persona-mount.html">3554</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/SpinButton-mount_base.html">2985</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26811/perf-test-react-components/SpinButton-mount.html">2961</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-avatar-f0ea8fa4-052d-4a92-a148-acfeb327bc3f.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-avatar-f0ea8fa4-052d-4a92-a148-acfeb327bc3f.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 27 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26811&commitId=a6874d153bebc8c8e35c4cb7c5a44baa1df2ae68&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1821 screenshots unchanged, 0 screenshots with different dimensions and 27 screenshots with visible difference.


  <details><summary><b>unknown</b> 27 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|AvatarGroup Converged.basic - Dark Mode.default.chromium.png|29789| Changed|
|AvatarGroup Converged.basic - High Contrast.default.chromium.png|55643| Changed|
|AvatarGroup Converged.basic.default.chromium.png|29248| Changed|
|AvatarGroup Converged.layoutPie - Dark Mode.default.chromium.png|9808| Changed|
|AvatarGroup Converged.layoutPie - High Contrast.default.chromium.png|21623| Changed|
|AvatarGroup Converged.layoutPie - RTL.default.chromium.png|12998| Changed|
|AvatarGroup Converged.layoutPie-1 - Dark Mode.default.chromium.png|2493| Changed|
|AvatarGroup Converged.layoutPie-1 - High Contrast.default.chromium.png|2421| Changed|
|AvatarGroup Converged.layoutPie-1 - RTL.default.chromium.png|3113| Changed|
|AvatarGroup Converged.layoutPie-1.default.chromium.png|3111| Changed|
|AvatarGroup Converged.layoutPie-2 - Dark Mode.default.chromium.png|2526| Changed|
|AvatarGroup Converged.layoutPie-2 - High Contrast.default.chromium.png|2389| Changed|
|AvatarGroup Converged.layoutPie-2 - RTL.default.chromium.png|3084| Changed|
|AvatarGroup Converged.layoutPie-2.default.chromium.png|3099| Changed|
|AvatarGroup Converged.layoutPie.default.chromium.png|12992| Changed|
|AvatarGroup Converged.layoutStack - Dark Mode.default.chromium.png|27709| Changed|
|AvatarGroup Converged.layoutStack - High Contrast.default.chromium.png|49959| Changed|
|AvatarGroup Converged.layoutStack.default.chromium.png|26665| Changed|
|AvatarGroup Converged.overflowContent - Dark Mode.chromium.png|7306| Changed|
|AvatarGroup Converged.overflowContent - Dark Mode.popoverContentOpen.chromium.png|49884| Changed|
|AvatarGroup Converged.overflowContent - High Contrast.chromium.png|8965| Changed|
|AvatarGroup Converged.overflowContent - High Contrast.popoverContentOpen.chromium.png|59364| Changed|
|AvatarGroup Converged.overflowContent.chromium.png|1003| Changed|
|AvatarGroup Converged.overflowContent.popoverContentOpen.chromium.png|6275| Changed|
|AvatarGroup Converged.overflowIndicator - Dark Mode.default.chromium.png|28163| Changed|
|AvatarGroup Converged.overflowIndicator - High Contrast.default.chromium.png|53717| Changed|
|AvatarGroup Converged.overflowIndicator.default.chromium.png|27585| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts [hunk 2]
```diff
@@ -23,38 +23,81 @@ const animations = {
...
-  root: {
-    display: 'inline-block',
-    flexShrink: 0,
-    position: 'relative',
-    verticalAlign: 'middle',
-    ...shorthands.borderRadius(tokens.borderRadiusCircular),
-    fontFamily: tokens.fontFamilyBase,
-    fontWeight: tokens.fontWeightSemibold,
-  },
+const avatarBorderRadius = '--fui-Avatar__borderRadius';
```
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts:26]: I don't think it really adds any complexity and based on the data we have

1. additional class names on a DOM node lead to a linear increase in style recalc time
2. local CSS vars are really cheap

this change is probably worth it just from a style recalc point of view. Taking into account that `shorthands.borderRadius()` must construct an object only to turn around and immediately destructure it, using shorthands is almost certainly slower when it comes to JS run time as well.

Of course we're talking about a very small number of things running once per component so we're definitely in micro-optimization territory.

Since this is an implementation detail and can be changed later I'll roll this back for now.

[the remark this answers is not in the document]
Comment (behowell) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts]: (Unresolving for visibility of my reply, but please re-resolve)

As far as I know, the shorthands destructuring happens in a pre-processing step by the griffel babel plugin, so that shouldn't impact runtime. If it does have a runtime impact, then we should look into changing shorthands in general, because we use it everywhere.

My main reason for my objection is that this is complicating the default and most common case of a round avatar, in exchange for a small perf win on the uncommon case of a square avatar.

In reality, the perf difference is likely negligible. But that's more reason to not introduce more code complexity for it :)

File: packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts [hunk 3]
```diff
@@ -36,8 +36,8 @@ export const useAvatarGroupStyles_unstable = (state: AvatarGroupState): AvatarGr
   state.root.className = mergeClasses(
     avatarGroupClassNames.root,
     styles.base,
-    layout === 'pie' && sizeStyles[size],
     layout === 'pie' && styles.pie,
+    layout === 'pie' && sizeStyles[size],
```
Comment (behowell) [inline packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts:40]: Is this change still needed?

File: packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItemStyles.ts [hunk 4]
```diff
@@ -29,6 +29,8 @@ const useRootStyles = makeStyles({
     ...shorthands.padding(tokens.spacingVerticalXS, tokens.spacingHorizontalXS),
   },
   nonOverflowItem: {
+    width: '32px',
+    height: '32px',
```
Comment (behowell) [inline packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItemStyles.ts:33]: Ditto: is this still needed?

File: packages/react-components/react-avatar/src/components/AvatarGroupPopover/useAvatarGroupPopoverStyles.ts [hunk 5]
```diff
@@ -167,8 +167,8 @@ export const useAvatarGroupPopoverStyles_unstable = (state: AvatarGroupPopoverSt
   state.triggerButton.className = mergeClasses(
     avatarGroupPopoverClassNames.triggerButton,
     groupChildClassName,
-    sizeStyles[size],
     triggerButtonStyles.base,
+    sizeStyles[size],
```
Comment (behowell) [inline packages/react-components/react-avatar/src/components/AvatarGroupPopover/useAvatarGroupPopoverStyles.ts:171]: Ditto: is this still needed?

File: packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts [hunk 6]
```diff
@@ -15,33 +15,35 @@ export const personaClassNames: SlotClassNames<PersonaSlots> = {
 
 const avatarSpacing = `--fui-Persona__avatar--spacing`;
 
+const useRootStyles = makeResetStyles({
```
Comment (behowell) [inline packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts:18]:
```suggestion
const useRootClassName = makeResetStyles({
```

File: packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts [hunk 7]
```diff
@@ -15,33 +15,35 @@ export const personaClassNames: SlotClassNames<PersonaSlots> = {
...
-    gridTemplateColumns: 'max-content [middle] auto',
+    // Intentionally empty
   },
   before: {
-    gridAutoFlow: 'column',
     justifyItems: 'end',
     gridTemplateColumns: 'auto [middle] max-content',
   },
   below: {
+    gridAutoFlow: 'unset',
```
Comment (behowell) [inline packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts:44]: nit: for clarity
```suggestion
    gridAutoFlow: 'row',
```

File: packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts [hunk 8]
```diff
@@ -204,17 +207,19 @@ export const usePersonaStyles_unstable = (state: PersonaState): PersonaState =>
...
-  base: {
-    display: 'block',
-  },
-  primaryText: {
-    color: tokens.colorNeutralForeground1,
-  },
-  optionalText: {
-    color: tokens.colorNeutralForeground2,
-  },
+const usePrimaryTextBaseStyles = makeResetStyles({
```
Comment (behowell) [inline packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts:210]:
```suggestion
const usePrimaryTextBaseClassName = makeResetStyles({
```

File: packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts [hunk 9]
```diff
@@ -204,17 +207,19 @@ export const usePersonaStyles_unstable = (state: PersonaState): PersonaState =>
...
+  display: 'block',
+  color: tokens.colorNeutralForeground1,
+  ...typographyStyles.body1,
+});
+
+const useOptionalTextBaseStyles = makeResetStyles({
```
Comment (behowell) [inline packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts:216]:
```suggestion
const useOptionalTextBaseClassName = makeResetStyles({
```

**Review by behowell — APPROVED** [review]

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts]: > As far as I know, the shorthands destructuring happens in a pre-processing step by the griffel babel plugin, so that shouldn't impact runtime.

Not be default, unfortunately, and I'm learning it can be difficult to enable the Webpack plugin in large, existing applications.

File: packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts (hunk 3 above)
Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts:40]: nope!

File: packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts (hunk 6 above)
Comment (spmonahan) [reply packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts:18]: Hmmm...I think something got mixed up because this change should not be in this PR. Will back these changes out of this one.

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (khmakoto) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts]: @spmonahan have you tried enabling the babel plugin instead of the webpack one? I find that it might be easier to do since it is easier to scope for large, existing applications. Allowing you to enable incrementally and, once you do that for the whole application, then allowing you to easily switch to the webpack plugin.

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts [hunk 10]
```diff
@@ -378,24 +378,27 @@ const useColorStyles = makeStyles({
...
 
+  const rootClassName = useRootClassName();
+  const imageClassName = useImageClassName();
+  const iconInitialsClassName = useIconInitialsClassName();
   const styles = useStyles();
   const sizeStyles = useSizeStyles();
   const colorStyles = useColorStyles();
 
-  const rootClasses = [styles.root, sizeStyles[size], colorStyles[color]];
+  const rootClasses = [rootClassName, size !== 32 && sizeStyles[size], colorStyles[color]];
```
Comment (khmakoto) [inline packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts:388]: You could alternatively alter line 202 to be

`32: { /* Default size in useRootClassName */ },`

and leave this line as

```suggestion
  const rootClasses = [rootClassName, sizeStyles[size], colorStyles[color]];
```

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts (hunk 10 above)
Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts:388]: Those styles are used in other avatar controls so it's simpler to leave it as-is.

File: packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts]: First off, I was mistaken and Fluent does have the Babel plugin enabled for its build so we're good there.

@khmakoto, the [docs state the Babel plugin is for libraries and the Webpack loader is for applications](https://github.com/microsoft/griffel/tree/main/packages/webpack-loader#when-to-use-it) so that's what I've gone with.

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by sopranopillow — APPROVED** [review]
